File: tuikit/src/main/java/com/tencent/qcloud/tim/uikit2/modules/conversation/holder/ConversationBaseHolder.java
package com.tencent.qcloud.tim.uikit2.modules.conversation.holder;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import com.tencent.qcloud.tim.uikit2.modules.conversation.ConversationListAdapter;
import com.tencent.qcloud.tim.uikit2.modules.conversation.base.ConversationInfo;

public abstract class ConversationBaseHolder extends RecyclerView.ViewHolder {

    protected View rootView;
    protected ConversationListAdapter mAdapter;

    public ConversationBaseHolder(View itemView) {
        super(itemView);
        rootView = itemView;
    }

    public void setAdapter(RecyclerView.Adapter adapter) {
        mAdapter = (ConversationListAdapter) adapter;
    }

    public abstract void layoutViews(ConversationInfo conversationInfo, int position);

}

File: im/src/main/java/com/black/im/adapter/holders/ConversationBaseHolder.kt
package com.black.im.adapter.holders

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.black.im.adapter.ConversationListAdapter
import com.black.im.model.ConversationInfo

abstract class ConversationBaseHolder(protected var rootView: View) : RecyclerView.ViewHolder(rootView) {
    protected var mAdapter: ConversationListAdapter? = null
    fun setAdapter(adapter: RecyclerView.Adapter<*>?) {
        mAdapter = adapter as ConversationListAdapter?
    }

    abstract fun layoutViews(conversationInfo: ConversationInfo?, position: Int)

}